libc-bottom-half/sources/getentropy.c

```diff
@@ -1,18 +1,18 @@
 #include <wasi/api.h>
+#include <pthread.h>
 #include <errno.h>
 #include <unistd.h>
 
-#ifdef _REENTRANT
-#error With threads support, getentropy is not intended to be a cancellation point.
-#endif
-
 int __getentropy(void *buffer, size_t len) {
+    int cs;
     if (len > 256) {
         errno = EIO;
         return -1;
     }
 
+    pthread_setcancelstate(PTHREAD_CANCEL_DISABLE, &cs);
     int r = __wasi_random_get(buffer, len);
+    pthread_setcancelstate(cs, 0);
```

Comment:
I propose that we leave out considerations of cancellation out for now. Pthread cancellation is something of an anti-sweet-spot, being particularly difficult and expensive to implement, while being particularly rarely used.

One of the particular complexities is what to do if one thread tries to cancel another thread that's currently executing host code. That might require all host code to be pthread cancel-safe, which would be very difficult for many popular host implementations today.

Reply:
Sounds reasonable, at least for a first pass.   How about we just avoid using `pthread_setcancelstate` et al in the bottom half and just stub them out (make them no-ops) when compiling top-half code.